pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -625,20 +643,32 @@ func (vs *VSphere) AttachDisk(vmDiskPath string, storagePolicyID string, nodeNam
 		if err != nil {
 			return "", err
 		}
-		vm, err := vs.getVMByName(ctx, nodeName)
+
+		vm, err := vs.getVMFromNodeName(ctx, nodeName)
 		if err != nil {
 			glog.Errorf("Failed to get VM object for node: %q. err: +%v", convertToString(nodeName), err)
 			return "", err
 		}
-		diskUUID, err = vm.AttachDisk(ctx, vmDiskPath, &vclib.VolumeOptions{SCSIControllerType: vclib.PVSCSIControllerType, StoragePolicyID: storagePolicyID})
+
+		diskUUID, err = vm.AttachDisk(ctx, vmDiskPath, &vclib.VolumeOptions{SCSIControllerType: vclib.PVSCSIControllerType, StoragePolicyName: storagePolicyName})
```

Comment:
Now that everything depends on storagepolicyName,  I guess you can remove storagePolicyID from volumeOptions.

Reply:
Dont we store volume options in PV? I remember seeing storage policy ID in the PV. If we remove it, how would upgrade work? If we dont want to use policy ID going forward, we should deprecate it.

Reply:
Nope, we don't need to store volume options in PV.  I am not saying to remove storagepolicyID in PV spec but rather remove it from internal struct volume options.